src/main/java/io/appium/java_client/AppiumDriver.java

```diff
@@ -129,6 +127,35 @@ public AppiumDriver(HttpClient.Factory httpClientFactory, Capabilities desiredCa
 
     public AppiumDriver(Capabilities desiredCapabilities) {
         this(AppiumDriverLocalService.buildDefaultService(), desiredCapabilities);
+
+    }
+
+    /**
+     * For creating a client connected to an existing Appium session.
+     * The other constructors all create a new session, sending the POST /session request.
+     * This constructor does not.
+     */
+    public AppiumDriver(URL remoteAddress, String sessionId) {
+        this.setSessionId(sessionId);
+        
+        AppiumCommandExecutor executor = new AppiumCommandExecutor(MobileCommand.commandRepository, remoteAddress);
+        executor.setCommandCodec(new AppiumW3CHttpCommandCodec());
```

Comment:
how do we know that this is a w3c sessions?

Reply:
I'd like to limit this feature to W3C sessions only. I had a nice error before but refactored it away, I'll find a way to surface an error if it's not W3C. Any idea how?

Reply:
the simplest way would be to get /status response and analyse its format. The response in W3C format does not contain response code field (the same is done in appium base driver)

Reply:
So the current code sets this from the initial response to create a session. The new code never sends that request, so it needs to be set. BUT, the new code calls GET /session/:sessionId when setting the JsonToMobile converter thing. So the codec needs to be set before the constructor finishes.

I could change the executor code to set this if the codecs are null, rather than just on create session. I'll try that.

Reply:
I would try to make the request directly using okhttp wrapper

Reply:
So I was going to change the execute method to set the codecs if they are currently null, rather than only if it's a create_session command. BUT, in order to send and encode any command, it needs codecs already. So there's no way to capture the protocol type when the first command is sent, because the command can't be sent :(

options I can think of:
 - make this a W3C-only feature
 - remove jsonwire protocol support in a major version
 - add a third term to the constructor, requiring the user pass in a dialect (I wanted to avoid this because it's an extra complication)

Any idea when we can stop supporting json wire?

Reply:
I ran the test with the `MobileCapabilityType.FORCE_MJSONWP` desired cap, and the test still passed.